src/ast.rs

```diff
@@ -0,0 +1,72 @@
+//! Provides a type system for the AST, in some sense
+
+mod expr_ext;
+mod node_ext;
+mod nodes;
+mod operators;
+mod token_ext;
+mod tokens;
+
+use crate::{
+    NixLanguage,
+    SyntaxKind::{self, *},
+    SyntaxToken,
+};
+
+pub use nodes::*;
+pub use operators::{BinOpKind, UnaryOpKind};
+pub use tokens::*;
+
+pub trait AstNode: rowan::ast::AstNode<Language = NixLanguage> {}
+
+impl<T> AstNode for T where T: rowan::ast::AstNode<Language = NixLanguage> {}
+
+mod support {
+    use rowan::ast::AstChildren;
+
+    use super::{AstNode, AstToken};
+    use crate::{SyntaxElement, SyntaxKind, SyntaxNode, SyntaxToken};
+
+    pub(super) fn first<N: AstNode, NN: AstNode>(parent: &N) -> Option<NN> {
```

Comment:
I think just using `pub` instead of `pub(super)` would look better. `mod support` isn't exported anywhere anyways, so it would make a difference anyways.